code/datums/lazy_template.dm

```diff
@@ -0,0 +1,76 @@
+
+/**
+ * Datum used to designate certain areas that do not need to exist nor be loaded at world start
+ * but do want to be loaded under certain circumstances. Use this for stuff like the nukie base or wizden, aka stuff that only matters when their antag is rolled.
+ */
+/datum/lazy_template
+	/// If this is true each load will increment an index keyed to the type and it will load [map_name]_[index]
+	var/list/datum/turf_reservation/reservations = list()
+	var/uses_multiple_allocations = FALSE
+	var/key
+	var/map_dir = "_maps/templates/lazy_templates"
+	var/map_name
+	var/map_width
+	var/map_height
+
+/datum/lazy_template/New()
+	reservations = list()
+	..()
+
+/datum/lazy_template/Destroy(force, ...)
+	if(!force)
+		return
+
+	QDEL_LIST(reservations)
+	GLOB.lazy_templates -= key
+	return ..()
+
+/**
+ * Does the grunt work of loading the template.
+ */
+/datum/lazy_template/proc/lazy_load()
+	RETURN_TYPE(/turf)
+	var/static/list/multiple_allocation_hash
+
+	var/load_path = "[map_dir]/[map_name].dmm"
+	if(uses_multiple_allocations)
+		var/times = LAZYACCESS(multiple_allocation_hash, type) || 0
```

Comment:
I wouldn't bother with the lazy list here and would just default to `list()`, makes the rest of this uglier